tasks/sierpinski-triangle/src/main.rs

```diff
@@ -1,6 +1,33 @@
 use std::iter::repeat;
 
+fn sierpinski(order: usize) {
+    let mut triangle = vec!["*".to_string()];
+    for i in 0..order {
+        let space = repeat(' ').take(2_usize.pow(i as u32)).collect::<String>();
+
+        // save original state
+        let mut d = triangle.clone();
+
+        // extend existing lines
+        d.iter_mut().for_each(|r| {
+            let new_row = format!("{}{}{}", space, r, space);
+            *r = new_row;
+        });
+
+        // add new lines
+        triangle.iter().for_each(|r| {
+            let new_row = format!("{}{}{}", r, " ", r);
+            d.push(new_row);
+        });
+
+        triangle = d;
+    }
+
+    triangle.iter().for_each(|r| println!("{}", r));
+}
 fn main() {
+    sierpinski(4);
+
     let order = 4;
```

Comment:
Let's remove the old code since it doesn't work.

I think it would be a good idea to write an integration test in this crate that uses [`assert_cmd`](https://docs.rs/assert_cmd/1.0.3/assert_cmd/) to test that the output exactly matches the expected triangle.